src/core/index.ts

```diff
@@ -246,27 +245,54 @@ export interface BannerPromptConfig extends BasePromptConfig, PromptUiActions {
     backgroundMask?: BackgroundMaskConfig;
 }
 
-export type PromptConfig =
+// DDL BANNER
+
+type DDLBannerLabelConfig = DialogLabelConfig<PromptTypeName.BANNER_DL>;
+type DDLBannerColorConfig = DialogColorConfig<PromptTypeName.BANNER_DL>;
+
+export interface DDLBannerPromptConfig
+    extends BasePromptConfig,
+        PromptUiActions {
+    type: PromptTypeName.BANNER_DL;
+    labels: DDLBannerLabelConfig;
+    colors: DDLBannerColorConfig;
+    imageUrl: string;
+    storeUrl: string;
+    canonicalLinkUrl: string;
+}
+
+export type PushPromptConfig =
     | BellPromptConfig
     | AlertPromptConfig
     | BannerPromptConfig;
 
-export type PromptConfigs = { [key: string]: PromptConfig };
+export type DDLPromptConfig = DDLBannerPromptConfig;
+
+export type PromptConfig = PushPromptConfig | DDLPromptConfig;
+
+export type PushPromptConfigs = Record<string, PushPromptConfig>;
+export type PromptConfigs = Record<string, PromptConfig>;
 
 export interface PlatformConfig {
-    publicKey: string;
+    publicKey?: string;
```

Comment:
Will this actually ever be optional? I appreciate in the config iface the user can provide, this field is now optional if they only consume DDL, but w.r.t. push platform config, this field would always be filled by the server.

Probably just missing something about the data flow, but if we could keep this more concrete that might be good.

Reply:
Yeah this was done when looking at the config load flow as it potentially can return {} - now, first thoughts were to make the config return null/undefined (preserving the typing on PlatformConfig but changing all the uses of loadConfig to look out for it) or make the type props optional (keeping the existing uses the same as they already could have {} in cases where the config just doesn't load) - I do prefer keeping the typing strict (as it was) and add a  little more sanity checking / guarding to handle failed config loads throughout the codebase... could look to the doing the latter, but had gone in this direction just due to existing code working with the possibility of getting an empty here...

Reply:
Hm, ok this possibly warrants further discussion. I would imagine if push is not enabled we can just bypass the entire flow that wants this config? Probs best pick up after everything else more finalised, let's chat.

Reply:
Turns out that loosened due to relaxing the `vapidPublicKey` however, this no longer needed to be the case (due to it being mapped from the api response) - nullable modifier now dropped back off from the prop.